### PR TITLE
No fail mode for DSC parser

### PR DESCRIPTION
**Pull request by matthewfcarlson**

### CODE REVIEW

File: edk2toollib/uefi/edk2/parsers/dsc_parser.py [hunk 1]
```diff
@@ -237,16 +238,35 @@ def ParseInfPathMod(self, line):
...
+                        raise
+                    else:
+                        self.Logger.warning(e)
 
     def __ProcessDefines(self, lines):
         if(len(lines) > 0):
             for l in lines:
-                (line, add) = self.__ParseDefineLine(l)
-                self.__ProcessDefines(add)
+                try:
```
Comment (spbrogan) [inline edk2toollib/uefi/edk2/parsers/dsc_parser.py:255]: why is this needed?

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dsc_parser.py:255]: Because we do include processing here. I was trying to catch the lines where we processed includes

Comment (spbrogan) [reply edk2toollib/uefi/edk2/parsers/dsc_parser.py:255]: why do we do it in two spots?

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dsc_parser.py:255]: Because the DSC parser is a two pass system

Comment (matthewfcarlson) [reply edk2toollib/uefi/edk2/parsers/dsc_parser.py:255]: It processes it once to find defines (including in includes), then does the proper parse. This is still a hold-over from the limited parser that was pre project mu

**Comment by matthewfcarlson** [comment]
@spbrogan how is this for comments?

**Comment by spbrogan** [comment]
Tested and works with OvmfPkg in Mu before and after dependencies have been populated.
